packages/admin/dashboard/src/routes/products/common/components/upload-media-form-item/upload-media-form-item.tsx

```diff
@@ -61,22 +64,37 @@ export const UploadMediaFormItem = ({
           }),
         })
 
-        return true
+        hasInvalidFile = true;
+      }
+
+      const fileSizeRejections = rejectedFiles.filter((f) => f?.reason === "size")
+
+      if (fileSizeRejections.length > 0) {
+        const fileNames = fileSizeRejections.map((f) => f.file.name).join(", ")
+        form.setError("media", {
```

Comment:
Q: what does it looks like with 10 files above the limits?

Reply:
<img width="662" height="545" alt="image" src="https://github.com/user-attachments/assets/93c520d1-0b9c-4f88-bf57-ff758ca37ac7" />

Reply:
do you think we should put new lines for each files? maybe only display the first 5 failing with `...` at the end?

Reply:
New lines could potentially result in a large list, I think a max number of errors could also be the better option. Should I create a new PR for this?

Reply:
Depending on the number of errors, it could potentially overflow on a new line if we don't set a max number of errors permitted to be displayed